nexusparser/tools/yaml2nxdl/yaml2nxdl_recursive_build.py

```diff
@@ -50,6 +59,9 @@ def xml_handle_exists(obj, k, v):
         raise ValueError('exists keyword found but value is None !')
 
 def xml_handle_dimensions(obj, k, v):
+    """
+    Xml attribute on existing object node in an xml tree
+    """
     if v != None:
```

Comment:
Also not part of this PR. 

Assert first and than do. This can get rid of most of the nesting:
```
if <bad>:
    raise ...
if <more-bad>:
    raise ...

<do the actual thing>
```

Is better than
```
if <good>:
   if <good>:
      <do>
   else:
      raise
else:
   raise
```